internal/services/oracle/oracle_exadata_infrastructure_resource.go

```diff
@@ -227,7 +245,12 @@ func (r ExadataInfraResource) Create() sdk.ResourceFunc {
 					CustomerContacts: pointer.To(ExpandCustomerContacts(model.CustomerContacts)),
 				},
 			}
-
+			if len(model.DatabaseServerType) > 0 {
+				param.Properties.DatabaseServerType = pointer.To(model.DatabaseServerType)
+			}
+			if len(model.StorageServerType) > 0 {
+				param.Properties.StorageServerType = pointer.To(model.StorageServerType)
+			}
```

Comment:
checks should be type specific:
```suggestion
			if model.DatabaseServerType != "" {
				param.Properties.DatabaseServerType = pointer.To(model.DatabaseServerType)
			}
			if model.StorageServerType != "" {
				param.Properties.StorageServerType = pointer.To(model.StorageServerType)
			}
```